src/plugins_exts/schema_mount.c

```diff
@@ -951,6 +956,202 @@ lyplg_ext_schema_mount_create_context(const struct lysc_ext_instance *ext, struc
     return res;
 }
 
+static void
+schema_mount_spriter_tree_free(void *priv)
+{
+    struct sprinter_tree_priv *st_priv;
+
+    st_priv = priv;
+    ly_set_free(st_priv->refs, NULL);
+    ly_ctx_destroy(st_priv->ext_ctx);
+    free(st_priv);
+}
+
+static LY_ERR
+schema_mount_sprinter_tree_cnode_override_mounted(const struct lysc_node *node, const void *UNUSED(plugin_priv),
+        ly_bool *UNUSED(filtered), char **UNUSED(flags), char **add_opts)
+{
+    if (!node->parent && add_opts) {
+        *add_opts = "/";
+    }
+
+    return LY_SUCCESS;
+}
+
+static LY_ERR
+schema_mount_sprinter_tree_pnode_override_mounted(const struct lysp_node *node, const void *UNUSED(plugin_priv),
+        ly_bool *UNUSED(filtered), char **UNUSED(flags), char **add_opts)
+{
+    if (!node->parent && add_opts) {
+        *add_opts = "/";
+    }
+
+    return LY_SUCCESS;
+}
+
+static LY_ERR
+schema_mount_sprinter_tree_node_override_parent_refs(const struct lysc_node *node, const void *plugin_priv,
+        ly_bool *filtered, char **UNUSED(flags), char **add_opts)
+{
+    uint32_t i;
+    const struct ly_set *refs;
+    const struct lysc_module *mod;
+    struct lysc_node *ref, *iter;
+
+    refs = ((struct sprinter_tree_priv *)plugin_priv)->refs;
+    mod = node->module->compiled;
+
+    /* Assume the @p node will be filtered. */
+    *filtered = 1;
+    for (i = 0; (i < refs->count) && *filtered; i++) {
+        ref = refs->snodes[i];
+        if (ref->module->compiled != mod) {
+            /* parent-reference points to different module */
+            continue;
+        }
+
+        for (iter = ref; iter; iter = iter->parent) {
+            if (iter == node) {
+                /* @p node is not filtered because it is parent-rererence node or his parent */
+                *filtered = 0;
+                break;
+            }
+        }
+    }
+
+    if (!*filtered && add_opts && !node->parent) {
+        /* top-node has additional opts */
+        *add_opts = "@";
+    }
+
+    return LY_SUCCESS;
+}
+
+static LY_ERR
+schema_mount_sprinter_ptree(struct lysp_ext_instance *UNUSED(ext), const struct lyspr_tree_ctx *UNUSED(ctx),
+        char **flags, char **UNUSED(add_opts))
+{
+    if (flags) {
```

Comment:
`flags` can be NULL? Please explain in the docs, the same for all the other `out` parameters.